remoulade/state/backends/redis.py

```diff
@@ -0,0 +1,51 @@
+import time
+from typing import Iterable
+
+import redis
+
+from ..backend import Missing, StateBackend
+
+
+class RedisBackend(StateBackend):
+
+    def __init__(self, *, namespace="remoulade-results", encoder=None, client=None, url=None, **parameters):
+        super().__init__(namespace=namespace, encoder=encoder)
+        if url:
+            parameters["connection_pool"] = redis.ConnectionPool.from_url(url)
+        self.client = client or redis.Redis(**parameters)
+        self.cancellation_ttl = 1000
+
+    def get_value(self, message_id):
+        """ Given a message_id returns the
+        value in the database"""
+        key = self.build_message_key(message_id)
+        print(key)
+        data = self.client.rpoplpush(key, key)
```

Comment:
The reason why we are using list on result backend is because we want to use the blocking operations of redis, here we do not care about this. You can use classic redis get/set here.